aplib.net-demo/Assets/Scripts/Configuration.cs

```diff
@@ -0,0 +1,277 @@
+    {
+        if (amount == "") amount = "0";
+        EnemyMeleeAmount = int.Parse(amount);
+        CheckAllValuesToBeValid();
+    }
+
+    /// <summary>
+    /// Set inputfield amount to variable.
+    /// Then call check to validate value.
+    /// </summary>
+    /// <param name="amount">Amount of ranged enemies.</param>
+    public void SetValueEnemyRanged(string amount)
+    {
+        if (amount == "") amount = "0";
+        EnemyRangedAmount = int.Parse(amount);
+        CheckAllValuesToBeValid();
+    }
+
+    /// <summary>
+    /// This is called when the grid values are changed.
+    /// This is also called when an individual component has changed.
+    /// It updates all other conponents.
+    /// </summary>
+    private void CheckAllValuesToBeValid()
+    {
+        // Check gridW and gridH
+        if (GridSizeX > MaxGridSize) GridSizeX = MaxGridSize;
+        else if (GridSizeX < MinGridSize) GridSizeX = MinGridSize;
+
+        if (GridSizeZ > MaxGridSize) GridSizeZ = MaxGridSize;
+        else if (GridSizeZ < MinGridSize) GridSizeZ = MinGridSize;
+
+        // Update value.
+        GridVolume = GridSizeX * GridSizeZ;
+
+        // Check room value.
+        if (RoomAmount > GridVolume * _maxPercentage) RoomAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (RoomAmount < 0) RoomAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemAmmoAmount value.
+        if (ItemAmmoAmount > GridVolume * _maxPercentage) ItemAmmoAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemAmmoAmount < 0) ItemAmmoAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemHealthAmount value.
+        if (ItemHealthAmount > GridVolume * _maxPercentage) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemHealthAmount < 0) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemRageAmount value.
+        if (ItemRageAmount > GridVolume * _maxPercentage) ItemRageAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemRageAmount < 0) ItemRageAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyMeleeAmount value.
+        if (EnemyMeleeAmount > GridVolume * _maxPercentage) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyMeleeAmount < 0) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyRangedAmount value.
+        if (EnemyRangedAmount > GridVolume * _maxPercentage) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyRangedAmount < 0) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check if items and enemies don't go over grid vol * _maxPrecentage(0.7f)
+        CheckItemOverflow();
+        CheckEnemyOverflow();
+
+        // Print correct number in text field. Skip this when testing.
+        if (!_test)
```

Comment:
Adding separate logic for tests is not really a good practice, but I could overlook it for now since we are short on time. This should be avoided though

Reply:
I thought so too, the playtest otherwise would be too long (I think) with setting all TMP's correctly, while it doesn't really matter.